Mage.Sets/src/mage/cards/g/GrimGiganotosaurus.java

```diff
@@ -0,0 +1,89 @@
+package mage.cards.g;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BecomesMonstrousSourceTriggeredAbility;
+import mage.abilities.costs.CostAdjuster;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.DestroyAllEffect;
+import mage.abilities.keyword.MonstrosityAbility;
+import mage.constants.ComparisonType;
+import mage.constants.SubType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.common.FilterOpponentsCreaturePermanent;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.filter.predicate.mageobject.PowerPredicate;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class GrimGiganotosaurus extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent("artifacts and creatures other than Grim Giganotosaurus");
+    static {
+        filter.add(AnotherPredicate.instance);
+        filter.add(Predicates.or(CardType.ARTIFACT.getPredicate(), CardType.CREATURE.getPredicate()));
+    }
+
+    public GrimGiganotosaurus(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{B}{G}");
+        
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(10);
+        this.toughness = new MageInt(10);
+
+        // {10}{B}{G}: Monstrosity 10. This ability costs {1} less to activate for each creature with power 4 or greater your opponents control.
+        // Based on Nemesis of Mortals
+        int monstrosityValue = 10;
+        Ability ability = new MonstrosityAbility("{10}{B}{G}", monstrosityValue);
```

Comment:
I kinda want to refactor (target adjuster and string as constructor arguments) but it's probably not worth the trouble

Reply:
took a shot at it, lemme know if that isnt what you had in mind.

Reply:
nope, I meant for MonstrosityAbility to have a cost adjuster field that it uses if not null, to support this and nemesis

the way you changed it isn't really different from what you had at first, except it's missing a copy constructor that it needs since it's a subclass, so I don't think it's better

Reply:
I see--like an optional constructor to `MonstrosityAbility` to allow passing in a `CostAdjuster` and a `costAdjusterText` to do essentiually what is being done here for `GrimGigantosaurusAbility`?

Reply:
Yeah, that's what I was thinking

Reply:
Please write a simple unit test for Nemesis of Mortals to confirm functionality, and then refactor it to match this.

Reply:
(I think I was right about this not being worth the trouble)